sdk/cosmos/azure-cosmos/test/test_latest_session_token.py

```diff
@@ -0,0 +1,214 @@
+                                                   offer_throughput=400)
+        feed_ranges_and_session_tokens = []
+        previous_session_token = ""
+        pk = ['CA', 'LA1', '90001']
+        pk_feed_range = container.feed_range_from_partition_key(pk)
+        target_session_token, target_feed_range, previous_session_token = self.create_items_physical_pk(container,
+                                                                                                        pk_feed_range,
+                                                                                                        previous_session_token,
+                                                                                                        feed_ranges_and_session_tokens,
+                                                                                                        True)
+
+        session_token = container.get_latest_session_token(feed_ranges_and_session_tokens, target_feed_range)
+        assert session_token == target_session_token
+        self.database.delete_container(container.id)
+
+
+    def test_latest_session_token_logical_hpk(self):
+        container = self.database.create_container("test_updated_session_token_from_logical_hpk" + str(uuid.uuid4()),
+                                                   PartitionKey(path=["/state", "/city", "/zipcode"], kind="MultiHash"),
+                                                   offer_throughput=400)
+        feed_ranges_and_session_tokens = []
+        previous_session_token = ""
+        target_pk = ['CA', 'LA1', '90001']
+        target_feed_range = container.feed_range_from_partition_key(target_pk)
+        target_session_token, previous_session_token = self.create_items_logical_pk(container, target_feed_range,
+                                                                                    previous_session_token,
+                                                                                    feed_ranges_and_session_tokens,
+                                                                                    True)
+        session_token = container.get_latest_session_token(feed_ranges_and_session_tokens, target_feed_range)
+
+        assert session_token == target_session_token
+        self.database.delete_container(container.id)
+
+
+    @staticmethod
+    def trigger_split(container, throughput):
+        print("Triggering a split in session token helpers")
+        container.replace_throughput(throughput)
+        print("changed offer to 11k")
+        print("--------------------------------")
+        print("Waiting for split to complete")
+        start_time = time.time()
+
+        while True:
+            offer = container.get_throughput()
+            if offer.properties['content'].get('isOfferReplacePending', False):
+                if time.time() - start_time > 60 * 25:  # timeout test at 25 minutes
```

Comment:
slightly concerned about us adding yet another test reliant on performing a partition split/ adding potentially 25mins to every run but I understand the need